app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/WeatherTest.kt

```diff
@@ -0,0 +1,152 @@
+package com.monkeyteam.chimpagne.newtests
+
+import androidx.test.platform.app.InstrumentationRegistry
+import com.monkeyteam.chimpagne.model.location.GetWeatherConstants
+import com.monkeyteam.chimpagne.model.location.Location
+import com.monkeyteam.chimpagne.model.location.getWeather
+import java.time.LocalDate
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class WeatherTest {
+  // Using test api key
+  private lateinit var server: MockWebServer
+  private val apiKey = "test_api_key"
+  val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+  @Before
+  fun setUp() {
+    server = MockWebServer()
+    server.start()
+    GetWeatherConstants.PORT = server.port
+    GetWeatherConstants.SCHEME = "http"
+    GetWeatherConstants.HOST = server.hostName
+    GetWeatherConstants.VERSION = ""
+    GetWeatherConstants.FORECAST = ""
+  }
+
+  @After
+  fun tearDown() {
+    server.shutdown()
+  }
+
+  @Test
+  fun testSuccessfulResponse() {
+    val json =
+        """
```

Comment:
Next time, build the json in a nicer way next time. A huge string like this without syntax highlting is prone to errors.
Obviously, since this is just for a test, it's not a critical issue.